features/0270-protocol-test-suite/README.md

```diff
@@ -0,0 +1,225 @@
+
+Without careful effort, though, interoperability is subjective and slippery. If products A and B implement the same spec, or if they demo cooperation in a single workflow, does that mean they can be used together? How much? For how long? Across which release boundaries? With what feature caveats?
+
+We need a methodology that gives crisp answers to questions like these--and it needs to be more efficient than continuously exercising every feature of every product against features of every other product.
+
+However, it's important to temper our ambitions. Standards, community specs, and reference implementations exist, and many of them come with tests or test suites of their own. Products can test themselves with these tools, and with custom tests written by their dev staffs, and make rough guesses about interoperability. The insight we're after is a bit different.
+
+### Goals
+
+What we need is a tool that achieves these goals:
+
+1. **Evaluate practical interoperability** of [agents](../../concepts/0004-agents/README.md).
+
+    >[Other software that offers SSI features](../../concepts/0004-agents/README.md#the-agent-ness-continuum) should also be testable. Here, such components are conflated with agents for simplicity, but it's understood that the suite targets protocol participants no matter what their technical classification.
+ 
+   Focus on remote interactions that deliver business value: [high-level protocols](../../concepts/0003-protocols/README.md) built atop [DIDComm](../../concepts/0005-didcomm/README.md), such as [credential issuance](../0036-issue-credential/README.md), [proving](../0037-present-proof/README.md), and [introducing](../0028-introduce/README.md), where each [participant](../../concepts/0003-protocols/roles-participants-etc.md#participants) uses different software. DID methods, ledgers, crypto libraries, credential implementationss, and DIDComm infrastructure should have separate tests that are out of scope here. None of these generate deep insight into whether packaged software is interoperable enough to justify purchase decisions; that's the gap we need to plug.
+   
+2. **Describe results in a formal, granular, reproducible way** that supports comparison between agents A and B, and between A at two different points of time or in two different configurations.
+
+    This implies a structured report, as well as support for versioning of the suite, the agents under test, and the results.
+
+3. **Track the collective community state of the art**, so measurements are comprehensive and up-to-date, and so new ideas automatically encounter pressure to be vetted for interoperability.
+ 
+    The test suite isn't a compliance tool, and it should be unopinionated about what's important and what's not. However, it should embody a broad list of testable features--[certainly, ones that are standard](/README.md#accepted), and often, [ones that are still maturing](/README.md#demonstrated).
```

Comment:
> The test suite isn't a compliance tool

This is an odd statement to include in an RFC for a test suite with the ultimate goal of ensuring interoperability.

Reply:
My understanding of this statement is that he's saying there's no set of required protocols that must be supported in order to be "compliant". E.g. an implementer may choose to not support the trust ping protocol, which doesn't make them non-compliant.